### PR TITLE
NBSNEBIUS-86: fixed stupid bug: FilteredFreshDeviceIds filtration should not drop all FreshDeviceIds for volumes created after 2023-08-30

### PR DESCRIPTION
**Pull request by qkrorlqr**
I introduced this bug in https://github.com/ydb-platform/nbs/commit/9c45422d82a3cc2ff4791b117b286b5cb4c5edb7 (2 days ago), haven't merged it to stable-23-3 yet, caught it in cloud/blockstore/tests/loadtest/local-mirror mirror2-agent-removal test.

### CODE REVIEW

**Review by debnatkh — APPROVED** [review]

**Comment by qkrorlqr** [comment]
FAILED 1: cloud/blockstore/tools/testing/eternal_tests/eternal-load/bin/tests/test.py.test_load_fails
этот тест флапает

File: cloud/blockstore/libs/storage/volume/volume_state.cpp [hunk 1]
```diff
@@ -266,6 +266,10 @@ void TVolumeState::Reset()
         for (const auto& r: Meta.GetReplicas()) {
             addFreshDevices(r.GetDevices());
         }
+    } else {
+        FilteredFreshDeviceIds = THashSet<TString>(
+            Meta.GetFreshDeviceIds().begin(),
```
Comment (drbasic) [inline cloud/blockstore/libs/storage/volume/volume_state.cpp:271]: Да, понятно почему ошибка проскочила. У лямбды область видимости должна быть уже.
А еще лучше вынести в отдельный метод получение FilteredFreshDeviceIds.
И тогда будет FilteredFreshDeviceIds = GetFilteredFreshDeviceIds();
А внутри GetFilteredFreshDeviceIds будет простой, понятный код с одним if 
```
if (GetCreationTs() > oldDate) {
return THashSet<TString>(
            Meta.GetFreshDeviceIds().begin(), Meta.GetFreshDeviceIds().end());
}
а тут сложная логика + return.

```

**Review by drbasic — APPROVED** [review]

File: cloud/blockstore/libs/storage/volume/volume_state.cpp [hunk 2]
```diff
@@ -244,33 +244,39 @@ void TVolumeState::Reset()
...
+                device.GetDeviceUUID()) != ids.end();
 
             if (found) {
-                FilteredFreshDeviceIds.insert(device.GetDeviceUUID());
+                filtered.insert(device.GetDeviceUUID());
             }
         }
     };
 
-    // XXX the aforementioned bug affects only "old" disks, see NBS-4383
```
[the remark this answers is not in the document]
Comment (qkrorlqr) [reply cloud/blockstore/libs/storage/volume/volume_state.cpp:262]: тут случайно потерялся, но вообще он есть в юниттесте, так что пофиг

**Comment by qkrorlqr** [comment]
TEST NAME	ELAPSED	STATUS	LOG
cloud/disk_manager/internal/pkg/clients/nbs/tests/sole chunk chunk	0.000s	FAIL	[LOG](https://github-actions-s3.website.nemax.nebius.cloud/ydb-platform/nbs/PR-check/7666924933/1/nebius-x86-64/logs/cloud/disk_manager/internal/pkg/clients/nbs/tests/test-results/gofmt/run_test.log)
cloud/disk_manager/internal/pkg/clients/nbs/tests/tests.TestCloneDiskFromOneZoneToAnother	20.000s	FAIL	[LOG](https://github-actions-s3.website.nemax.nebius.cloud/ydb-platform/nbs/PR-check/7666924933/1/nebius-x86-64/logs/cloud/disk_manager/internal/pkg/clients/nbs/tests/test-results/tests/testing_out_stuff/tests.TestCloneDiskFromOneZoneToAnother.log)
cloud/disk_manager/internal/pkg/clients/nbs/tests/tests.TestFinishFillDisk	20.010s	FAIL	[LOG](https://github-actions-s3.website.nemax.nebius.cloud/ydb-platform/nbs/PR-check/7666924933/1/nebius-x86-64/logs/cloud/disk_manager/internal/pkg/clients/nbs/tests/test-results/tests/testing_out_stuff/tests.TestFinishFillDisk.log)
cloud/disk_manager/internal/pkg/clients/nbs/tests/tests.TestGetChangedBlocksForLightCheckpoints	20.010s	FAIL	[LOG](https://github-actions-s3.website.nemax.nebius.cloud/ydb-platform/nbs/PR-check/7666924933/1/nebius-x86-64/logs/cloud/disk_manager/internal/pkg/clients/nbs/tests/test-results/tests/testing_out_stuff/tests.TestGetChangedBlocksForLightCheckpoints.log)
cloud/disk_manager/internal/pkg/clients/nbs/tests/tests.TestGetChangedBytes	31.040s	FAIL	[LOG](https://github-actions-s3.website.nemax.nebius.cloud/ydb-platform/nbs/PR-check/7666924933/1/nebius-x86-64/logs/cloud/disk_manager/internal/pkg/clients/nbs/tests/test-results/tests/testing_out_stuff/tests.TestGetChangedBytes.log)

флапы

**Pull request merged**